### PR TITLE
fix(chat): add confirmation dialog for code editor exit without saving (Issue #2734)

### PR DESCRIPTION
**Pull request by Gimir**
**Description:**

- add OptionsModal component
- remove old application modal components
- add save all action for code editor

Issues:

- Issue #2734 

**UI changes**
Снимок экрана 2024-12-17 в 19 35 54

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by Gimir** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2830.nightly-test.deltixhub.io
>E2E tests status: success

**Review by denys-kolomiitsev — DISMISSED** [review]

File: apps/chat/src/components/Common/ApplicationWizard/CodeAppView/CodeAppView.tsx [hunk 1]
```diff
@@ -313,6 +371,13 @@ export const CodeAppView: FC<ViewProps> = ({
           />
         </div>
 
+        <OptionsDialog
+          isOpen={!!editorConfirmation}
+          heading={t('Do you want to save changes in code editor?')}
```
Comment (IlyaBondar) [inline apps/chat/src/components/Common/ApplicationWizard/CodeAppView/CodeAppView.tsx]:
```suggestion
          heading={t('Do you want to save changes in the code editor?')}
```

**Comment by Gimir** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2830.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/components/Common/OptionsDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  const descriptionId = useId();
+
+  return (
+    <Modal
+      portalId="theme-main"
+      state={isOpen ? ModalState.OPENED : ModalState.CLOSED}
+      onClose={onClose ?? fakeCloseHandler}
+      dataQa="options-dialog"
+      containerClassName="inline-block w-full min-w-[90%] px-3 py-4 md:p-6 text-center md:min-w-[300px] md:max-w-[500px]"
+      dismissProps={{ outsidePressEvent: 'mousedown', outsidePress: true }}
```
Comment (IlyaBondar) [inline apps/chat/src/components/Common/OptionsDialog.tsx]: it's new object each time. Let's move it in modal constants.
I suggest to create in constants file and use everywhere (see `modalDismissProps`):
- const MOUSE_OUTSIDE_PRESS_EVENT = { outsidePressEvent: 'mousedown' }
- const OUTSIDE_PRESS = { outsidePress: true }
- const OUTSIDE_PRESS_AND_MOUSE_EVENT = { ...MOUSE_OUTSIDE_PRESS_EVENT, ...OUTSIDE_PRESS  }

**Review by IlyaBondar — DISMISSED** [review]

**Comment by Gimir** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2830.nightly-test.deltixhub.io
>E2E tests status: success

**Review by IlyaBondar — APPROVED** [review]

**Pull request merged**